File: app/src/main/java/com/brianmk/exposurenotes/CameraSettingsDialog.java
package com.brianmk.exposurenotes;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;

public class CameraSettingsDialog extends DialogFragment {
    private static final String LOG_TAG = CameraSettingsDialog.class.getSimpleName();

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        View content = getActivity().getLayoutInflater().inflate(R.layout.camera_settings_dialog, null);
        builder.setView(content);

        // Film formats
        Spinner formatSpinner = content.findViewById(R.id.default_spinner);
        ArrayAdapter<CharSequence> activityAdapter = ArrayAdapter.createFromResource(content.getContext(),
                R.array.format_types, R.layout.default_spinner_item);
        activityAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        formatSpinner.setAdapter(activityAdapter);
        formatSpinner.setSelection(0);

        // Save button
        Button saveButton = content.findViewById(R.id.save_button);
        saveButton.setOnClickListener(new AdapterView.OnClickListener() {
            @Override
            public void onClick(View v) {
                // save data stuff goes here
                dismiss();
            }
        });

        // Cancel button
        Button cancelButton = content.findViewById(R.id.cancel_button);
        cancelButton.setOnClickListener(new AdapterView.OnClickListener() {
            @Override
            public void onClick(View v) {
                dismiss();
            }
        });

        return builder.create();
    }
}

File: app/src/main/java/com/brianmk/exposurenotes/CameraSettingsDialog.kt
package com.brianmk.exposurenotes

import android.app.AlertDialog
import android.app.Dialog
import android.app.DialogFragment
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.Spinner

class CameraSettingsDialog : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle): Dialog {
        val builder = AlertDialog.Builder(activity)
        val content = activity.layoutInflater.inflate(R.layout.camera_settings_dialog, null)
        builder.setView(content)

        // Film formats
        val formatSpinner = content.findViewById<Spinner>(R.id.default_spinner)
        val activityAdapter = ArrayAdapter.createFromResource(content.context,
                R.array.format_types, R.layout.default_spinner_item)
        activityAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        formatSpinner.adapter = activityAdapter
        formatSpinner.setSelection(0)

        // Save button
        val saveButton = content.findViewById<Button>(R.id.save_button)
        saveButton.setOnClickListener( {
            fun onClick(v: View) {
                // save data stuff goes here
                dismiss()
            }
        })

        // Cancel button
        val cancelButton = content.findViewById<Button>(R.id.cancel_button)
        cancelButton.setOnClickListener( {
            fun onClick(v: View) {
                dismiss()
            }
        })

        return builder.create()
    }

    companion object {
        private val LOG_TAG = CameraSettingsDialog::class.java.simpleName
    }
}
